Bunch of features and fixes for the window picker applet

milouse opened the pull request:
This PR brings some love to the window picker applet:

- remove some compilation warnings
- remove runtime warning in `~/.xsession-errors` file
- add back missing settings
- [NEW] allow one to unboldify app title

This PR may also be seen as a clean working base for #39 

As a new comer to the Mate project, I'll take care of all your reviews. Do not hesitate to comment if you think I don't follow any guideline you may have. Thank you for your understanding.

raveit65 commented:
Can you please edit commits (not add a new one) and remove changes in language files (.mo)?
We use transifex sever for translations. https://www.transifex.com/mate/MATE/dashboard/
Every change here at github will be override there.
Changing the resource file (.pot) is OK.

raveit65 commented on mate-window-picker-applet/task-title.c:
```diff
@@ -49,15 +48,66 @@ struct _TaskTitlePrivate
+  GdkAppLaunchContext *launch_context;
+  GdkDisplay *display;
+  GdkScreen *gdkscreen;
+  TaskTitlePrivate *priv;
```
Build warning.
https://travis-ci.org/mate-desktop/mate-netbook/jobs/521411788#L1175
```
task-title.c:69:21: warning: variable 'priv' set but not used [-Wunused-but-set-variable]
   TaskTitlePrivate *priv;
                     ^~~~
```

milouse replied:
Oh, sorry about that, it is a forgotten variable. I've fixed it in my last force-push.

raveit65 commented:
I don't build this applet for fedora, in result i can't do a runtime check.
So this needs to tested by someone who build and maintain it for his distro.
Maybe the ubuntu-mate maintainer?
@flexiondotorg

milouse commented:
> Can you please edit commits (not add a new one) and remove changes in language files (.mo)? […]
> Changing the resource file (.pot) is OK.

Done.

milouse commented:
Sorry for the delay, I was off all last week. I just came back and address your comments @raveit65. Thank you very much for your review!

monsta commented on mate-window-picker-applet/task-title.c:
```diff
@@ -454,27 +572,17 @@ task_title_init (TaskTitle *title)
-  priv->quit_icon = gtk_icon_theme_load_icon (theme, "mate-logout", width, 0, NULL);
-
+  gtk_icon_size_lookup (GTK_ICON_SIZE_MENU, &width, &height);
+  priv->quit_icon = gtk_icon_theme_load_icon (theme, "gnome-logout", width, 0, NULL);
```
Suggestion: use `system-log-out` icon name, as in mate-session-manager.

milouse replied:
Yes good idea. I didn't find this icon at first sight. I just saw that "mate-logout" didn't really exists, that's why I replace it with gnome-logout. I've made the change and push again.

monsta commented:
In a quick test everything seems to work. I don't normally use this applet though, so I could miss something.

sc0w commented:
cppcheck warnings:

```
[mate-window-picker-applet/task-title.c:66]: (style) The scope of the variable 'launch_context' can be reduced.
[mate-window-picker-applet/task-title.c:67]: (style) The scope of the variable 'display' can be reduced.
```
file `mate-netbook.pot` changed in this PR, @raveit65 it is ok?

milouse commented:
I've addressed the cppcheck warning problem (on my side, I didn't have these warnings. What is your cppcheck command line to be sure?).

I've rebased my branch on current master too.

sc0w commented:
thanks,  the warnings I mentioned are gone

@milouse 
> What is your cppcheck command line to be sure?

`cppcheck --enable=warning,style,performance,portability,information,missingInclude .`

milouse commented:
Hello,

Thank you very much for the review.

> **Two close buttons**
> 
> From the screengrab below, both the old `x` close button and the themed buttons are shown. Perhaps only the themed buttons should be visible?
> 
> ![netbook-crop](https://user-images.githubusercontent.com/304639/57194904-5372c300-6f44-11e9-9500-c414b89f4c19.png)

That's very weird. What others applets do you use? I suspect the second range of button in your screenshot doesn't belong to this applet.

> 
> **Show desktop title and logout button**
> 
> Toggling this option doesn't take effect until you log out and back into a new session. Perhaps this setting should toggle the logout button directly and not require a restart?

This is weird to because on my computer the change is "immediate": you don't need to logout and log-in again but to change active workspace once. I'll look if I can make it directly, without having to change workspace.

> 
> **Changing Window Border**
> 
> If I open Appearance Preferences and _Customise..._, changing the Window Border theme is not directly reflected in the applet. A logout and log in is required.

Ok, that's because the icon is loaded once for all when the applet is started (at session start). I'll see to reload it each time it's called.

I'm not sure I can listen to a theme change to determine if I must change the icon or not. Do you have any idea on this?

Thanks again,

flexiondotorg commented:
@milouse 

> That's very weird. What others applets do you use? I suspect the second range of button in your screenshot doesn't belong to this applet.

My mistake, very sorry! Yes, [I did have another applet loaded](https://github.com/ubuntu-mate/mate-window-applets) which caused me to submit an incorrect review of your changes :disappointed: 

I've created a new panel just to test your changes and everything is working correctly.

milouse commented:
@flexiondotorg no problem :)

Now that it's merged, I'll continue work on #39
